includes/Pda/PdaCustomServer.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: gaupoit
+ * Date: 9/26/18
+ * Time: 19:40
+ */
+
+class PdaCustomServer extends Wpup_UpdateServer {
+	/**
+	 * Override metedata.
+	 *
+	 * @param array        $meta Metadata.
+	 * @param Wpup_Request $request Request object.
+	 *
+	 * @return array|void
+	 */
+	protected function filterMetadata( $meta, $request ) {
+		$meta = parent::filterMetadata( $meta, $request ); // TODO: Change the autogenerated stub.
+		unset( $meta['download_url'] );
+		return $meta;
+	}
+	
+	/**
+	 * Override action download
+	 *
+	 * @param Wpup_Request $request Request object.
+	 */
+	protected function actionDownload( Wpup_Request $request ) {
+		$this->exitWithError( 'Downloads are disable', 403 );
+	}
+	
+	/**
+	 * Override load package for
+	 *
+	 * @param Wpup_Request $request Request object.
+	 */
+	protected function loadPackageFor( $request ) {
+		parent::loadPackageFor( $request );
+	}
+	
+	/**
+	 * Override authorization checking
+	 *
+	 * @param mixed $request Request object.
+	 */
+	protected function checkAuthorization( $request ) {
+		$this->logRequest( $request );
+        $license = $request->headers->get( 'Authorization' );
+        $api = $request->headers->get( 'ApiCheckLicense' );
+
+        $curl = curl_init();
+
+        curl_setopt_array($curl, array(
+            CURLOPT_URL => $api,
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_ENCODING => "",
+            CURLOPT_MAXREDIRS => 10,
+            CURLOPT_TIMEOUT => 30,
+            CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+            CURLOPT_CUSTOMREQUEST => "POST",
+            CURLOPT_POSTFIELDS => "{\n    \"license\": \"$license\"\n}",
```

Comment:
should create a php array and use json encode instead. It helps remove \n \" for being more readable.